app/scripts/views/mixins/open-confirmation-email-mixin.js

```diff
@@ -0,0 +1,96 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+
+/**
+ * A view mixin that helps confirm* views show the "Open Gmail" button
+ * directly from FxA.
+ */
+define(function (require, exports, module) {
+  'use strict';
+
+  var BaseView = require('views/base');
+  var t = BaseView.t;
+
+  return {
+    events: {
+      'click #open-webmail': '_webmailTabOpened'
+    },
+
+    /**
+     * Get the webmail URL for the given email
+     *
+     * @param {string} email
+     * @returns {string}
+     */
+
+    _isGmail: function (email) {
+      return /@gmail\.com$/.test(email);
+    },
+
+    _isHotmail: function (email) {
+      return /@hotmail\.com$/.test(email);
+    },
+
+    _isYahoo: function (email) {
+      return /@yahoo\.com$/.test(email);
+    },
+
+    getWebmailLink: function (email) {
+      var linkToBeReturned;
+
+      if (this._isGmail(email)){
+        linkToBeReturned = 'https://mail.google.com/mail/u/?authuser=' + encodeURIComponent(email);
+      } else if (this._isHotmail(email)){
+        linkToBeReturned = 'https://hotmail.com';
+      } else if (this._isYahoo(email)){
+        linkToBeReturned = 'https://login.yahoo.com/config/login';
```

Comment:
for yahoo we should users to `https://mail.yahoo.com`.  If you send them to `https://login.yahoo.com/config/login` then it will ask them to login even if they are already logged in to their yahoo email